.github/workflows/bump-version-and-create-release.yaml

```diff
@@ -37,50 +35,64 @@ jobs:
           INITIAL_VERSION: ${{ env.INITIAL_VERSION }}
           WITH_V: true
 
+
+  update-version-file:
+    needs: compute-version
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+
       - name: Update version file with new version
         run: |
-          echo "New version: ${{ steps.bump_version.outputs.new_tag }}"
-          echo "VERSION=${{ steps.bump_version.outputs.new_tag }}" > version
+          echo "New version: ${{ needs.compute-version.outputs.new_tag }}"
+          echo "VERSION=${{ needs.compute-version.outputs.new_tag }}" > version
           git config --local user.name "github-actions[bot]"
           git config --local user.email "github-actions[bot]@users.noreply.github.com"
           git add version
-          git commit -m "chore: update version file to ${{ steps.bump_version.outputs.new_tag }}"
+          git commit -m "chore: update version file to ${{ needs.compute-version.outputs.new_tag }}"
           git push
 
       - name: Push new tag
         run: |
-          git tag ${{ steps.bump_version.outputs.new_tag }}
-          git push origin ${{ steps.bump_version.outputs.new_tag }}
+          git tag ${{ needs.compute-version.outputs.new_tag }}
+          git push origin ${{ needs.compute-version.outputs.new_tag }}
+
 
   create-release:
-    needs: bump-version
+    needs: [compute-version, update-version-file]
     runs-on: ubuntu-latest
     permissions:
       contents: write
     steps:
       - name: Check out the repository and pull the new tag
         uses: actions/checkout@v4
         with:
-          ref: ${{ needs.bump-version.outputs.new_tag }}
+          fetch-depth: 0
+          ref: ${{ needs.compute-version.outputs.new_tag }}
 
       - name: Build release packages
         uses: docker/build-push-action@v6
         with:
           context: .
           push: false
           tags: |
-            util_caching_release
+            release_builder
           target: release
 
       - name: Copy release packages
         run: |
           mkdir -p /tmp/artifacts/
-          docker run --rm -v /tmp/artifacts:/tmp/artifacts util_caching_release cp -r /release /tmp/artifacts/
+          docker run --rm -v /tmp/artifacts:/tmp/artifacts release_builder cp -r /release /tmp/artifacts/
```

Comment:
Not directly related, but I guess you can simply use `docker cp`

https://docs.docker.com/reference/cli/docker/container/cp/

Reply:
I know about this one but that only works for running containers. So we'd have to start the container, then run docker cp. My solution is a single command that's why I opted for that one.

Reply:
Ah, didn't know about that – again sth. learned :grin: